Makefile

```diff
@@ -1,19 +1,39 @@
+# Delete target on error.
+# https://www.gnu.org/software/make/manual/html_node/Errors.html#Errors
+# > This is almost always what you want make to do, but it is not historical
+# > practice; so for compatibility, you must explicitly request it
+.DELETE_ON_ERROR:
+
 # Global tasks.
 # =============================================================================
 PYTHON_VERSION := python3.10
+VIRTUAL_ENV := .venv
 
 ifeq ($(USE_VENV),1)
 	EXEC_CMD :=
 else
 	EXEC_CMD := docker exec -ti itou_django
 endif
 
-.PHONY: run clean cdsitepackages quality fix pylint
+.PHONY: run clean cdsitepackages quality fix pylint update-deps
 
 # Run Docker images
 run:
 	docker-compose up
 
+$(VIRTUAL_ENV): requirements/dev.txt
+	$(PYTHON_VERSION) -m venv $@
+	$@/bin/pip install -r $^
+	$@/bin/pip-sync $^
+	touch $@
+
+update-deps:
+	$(VIRTUAL_ENV)/bin/pip-compile --upgrade --allow-unsafe --generate-hashes -o requirements/base.txt requirements/base.in
```

Comment:
ne faudrait-il pas ajouter par précaution les *.txt générés dans le .gitignore ?

Reply:
Est-ce que tu peux préciser pourquoi ?

Reply:
Pardon. On a deux options:
- soit on commite les fichiers *.txt générés à chaque fois que l'on met à jour les dépendances, ce qui permet de freezer les dépendances de manière ferme et hashed => méthode recommandée par le npm flow par exemple
- soit on est "comme aujourd"hui" et on n'a que les fichiers IN, donc les versions mais pas les hashes et chacun régénère ses fichiers de dépendance localement.

En effet, le premier mode semble préférable. Donc je n'ai rien dit.